src/Execute/executeQuickInput.ts

```diff
@@ -0,0 +1,151 @@
+    const backends: vscode.QuickPickItem[] = Object.keys(globalBackendMap).map(label => ({ label }));
+    state.infer = await input.showQuickPick({
+      title,
+      step: 1,
+      totalSteps: 3,
+      placeholder: 'Pick Executor',
+      items: backends,
+      shouldResume: shouldResume
+    });
+    state.backend = globalBackendMap[state.infer.label];
+    return (input: MultiStepInput) => selectInputModel(input, state);
+  }
+
+  // TODO: Use quickPick window with fast grep child process
+  async function selectInputModel(input: MultiStepInput, state: Partial<State>) {
+    if (state.backend === undefined) {
+      throw Error('Backend is undefined');
+    }
+
+    let backendName: string = state.backend.name();
+    const executor = state.backend.executor();
+    let filter: { [name: string]: string[]} = {};
+    // List files which are filtered with executable extensions
+    if (executor) {
+      filter = {
+        backendName: executor.getExecutableExt()
+      };
+    }
+
+    const fileUri = await vscode.window.showOpenDialog({
+      title: `${title} (2/3)`,
+      canSelectMany: false,
+      openLabel: 'Select Model to Infer',
+      filters: filter
+    });
+    if(fileUri && fileUri[0]) {
+      state.modelPath = fileUri[0];
+    }
+    else {
+      console.log("No model has been selected");
+      state.modelPath = undefined;
+    }
+
+    return (input: MultiStepInput) => selectInputSpec(input, state);
+  }
+
+  // TODO: enable the backend-driven option steps by backend extension
+  async function selectInputSpec(input: MultiStepInput, state: Partial<State>) {
+    const inputSpecKeys: vscode.QuickPickItem[] = ["any", "non-zero", "positive"].map(label => ({ label }));
+
+    state.infer = await input.showQuickPick({
+      title,
+      step: 3,
+      totalSteps: 3,
+      placeholder: 'Select Random Input Spec',
+      items: inputSpecKeys,
+      shouldResume: shouldResume
+    });
+    state.inputSpec = state.infer.label;
+  }
+
+  async function runInference (state: State){
+    let cmd = state.backend.executor()?.runInference(state.modelPath.path, ["--input-spec", state.inputSpec]);
+    const ws = vscode.workspace.workspaceFolders ? vscode.workspace.workspaceFolders.map(f => f.uri.fsPath) : process.cwd();
+    const q = process.platform === 'win32' ? '\\' : '/';
+    let outFileName: string = ws[0].concat(`${q}${state.modelPath.path.split(q).slice(-1)[0]}.infer.log`);
+
+    await vscode.window.withProgress({
+      location: vscode.ProgressLocation.Notification,
+      title: "Inference Running!",
+      cancellable: true
+    }, (progress, token) => {
+      token.onCancellationRequested(() => {
+        console.log("User canceled the log running operation");
+      });
+      const p = new Promise((resolve, reject) => {
+        exec(cmd?.str() + ' > ' + outFileName, (error, stdout, stderr) => {
+          if (error) { return reject(error.message); }
```

Comment:
Q. This returns rejects() and below code doesn't return resolve(). Why does it return differently?

Reply:
I'm sorry but I couldn't catch your question. What does `below code` mean?

Reply:
This

```ts
if (error) {
  return reject(error.message);
}
resolve(stdout);
```

Reply:
(First of all, I'm not familiar with Promise and related expressions. Please let me know if there is any misunderstood.

`vscode.window.withProgress`'s second parameter `task` is a callback which returns a promise. And, `withProgress` returns the promise which is returned by `task`.
```ts
      const p = new Promise((resolve, reject) => {
        exec(cmd?.str() + ' > ' + outFileName, (error, stdout, stderr) => {
          if (error) { return reject(error.message); }
          resolve(stdout);
        });
      });
      return p;
```
So, each `reject` and `resolve` are handled by `withProgress(...).then(()=>{}, ()=>{})` as you can find in the line after `return p;`.
If the command emits any error, it will show error notification. On the other case of successful execution, it will show **Inference Succeed!**.

Reply:
Umm... I was just curious about the reason why you didn't choose below options.

(1)
```ts
if (error) return rejct();
else return resolve();
```

(2)
```ts
if (error) reject()
else resolve()
```

You mixed `return reject()` and `resolve()`. I just wanted to know the reason. 😀

Reply:
And it could be important because of defining the return type.

Reply:
Oh, I'm totally ignoring the `return`.. I didn't intended but the source I referred wrote this way. I'll apply the first way 😅